includes/class-wc-stripe-webhook-handler.php

```diff
@@ -820,7 +825,21 @@ public function process_webhook_refund_updated( $notification ) {
 						$note = sprintf( __( 'Refund canceled for %1$s - Refund ID: %2$s - Reason: %3$s', 'woocommerce-gateway-stripe' ), $amount, $refund_object->id, $refund_object->failure_reason );
 					}
 
-					$order->add_order_note( $note );
+					// Store the failure reason
+					if ( isset( $refund_object->failure_reason ) ) {
+						$order->update_meta_data( '_stripe_refund_failure_reason', $refund_object->failure_reason );
+					}
```

Comment:
I think we should have this update and the update above occur in all cases, as it's possible that multiple attempts are made over time, and the two values should stay in sync, especially when the reason is empty.
```suggestion
					$order->update_meta_data( '_stripe_refund_status', $refund_object->status );
					if ( isset( $refund_object->failure_reason ) ) {
						$order->update_meta_data( '_stripe_refund_failure_reason', $refund_object->failure_reason );
					} else {
						$order->delete_meta_data( '_stripe_refund_failure_reason' );
					}
```